flagship.go

```diff
@@ -184,12 +197,16 @@ func (s *featureStore) GetHash(ctx context.Context, key string, hashKey io.Reade
 	return GetHash(ctx, key, hashKey)
 }
 
-func (s *featureStore) Bool(ctx context.Context, key string) bool {
+func (s *featureStore) Bool(ctx context.Context, key string) (res bool) {
 	f, _, err := s.fetch(ctx)
 	if err != nil {
 		f = s.cachedFeatures
 	}
-	return f.Bool(key)
+	res = f.Bool(key)
+	if s.logger != nil {
```

Comment:
I noticed that with the ThrottleAllow method you extracted a separate function, but not for `Bool`. 

Potentially better to keep them similar?

Reply:
I didn't wanted to revolutionise the codebase too much
Bool was quite easy to extend and add the logging bit
ThrottleAllow was a different beast
That was my heuristic BTW

I can comply with the request if you confirm this is indeed what we want - please :)

Reply:
Agree with the thinking, lets keep it separate.